src/webviews/extension-side/dataviewer/kernelDataViewerDependencyImplementation.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { executeSilently } from '../../../kernels/helpers';
+import { IKernel } from '../../../kernels/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { IDataViewerDependencyService } from './types';
+import { pandasMinimumVersionSupported } from './constants';
+
+export const kernelGetPandasVersion =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function kernelPackaging(kernel: IKernel): '%conda' | '%pip' {
```

Comment:
@DonJayamanne might know best here. But this will use %pip for envs like poetry. In those cases should we just not do the install and ask them to install manually?

Reply:
Yes agreed,

Reply:
what does that mean? how do I catch environments like poetry?

Reply:
I think it's unlikely this will ever pick 'conda' either. When we have an IKernel, we're in a remote situation. Otherwise we'd be using the interpreter.

I think we can leave this as is for now, and then figure out some other way to detect the presence of the package manager later.